### PR TITLE
remove heavy dependency on boost foreach file

### PR DESCRIPTION
**Pull request by pwnorbitals**
Replaced all occurences of it with range-based for loops. Should reduce build times !

### CODE REVIEW

File: src/net.h [hunk 1]
```diff
@@ -22,6 +21,25 @@
...
+// Should be replaced by range-based for loops !
+namespace std {
+    template<class T> T begin(std::pair<T, T> p){return p.first;}
+    template<class T> T end(std::pair<T, T> p){return p.second;}
+}
+template<class Iterator> std::reverse_iterator<Iterator> make_reverse_iterator(Iterator it)
+{return std::reverse_iterator<Iterator>(it);}
+template<class Range> std::pair<std::reverse_iterator<decltype(begin(std::declval<Range>()))>,
+                      std::reverse_iterator<decltype(begin(std::declval<Range>()))>> make_reverse_range(Range&& r)
+{return std::make_pair(make_reverse_iterator(begin(r)), make_reverse_iterator(end(r)));}
```
Comment (tomasbrod) [inline src/net.h:37]: this and above Looks like some black magic

File: src/net.h [hunk 2]
```diff
@@ -378,7 +396,7 @@ class CNode
     unsigned int GetTotalRecvSize()
     {
         unsigned int total = 0;
-        BOOST_FOREACH(const CNetMessage &msg, vRecvMsg) 
+        BOOST_FOREACH(const CNetMessage &msg, vRecvMsg)
```
Comment (tomasbrod) [inline src/net.h:399]: accidental white-space change

**Review by tomasbrod — CHANGES_REQUESTED** [review]

**Comment by denravonska** [comment]
I'm not sure this is the way to go. It adds a layer and a bit of confusion. I think it's better to do these in steps. If we want to improve build speeds we can focus on the foreaches which reside in the headers.

File: src/net.h (hunk 1 above)
Comment (pwnorbitals) [reply src/net.h:37]: It does. Hopefully we'll get rid of it as soon as all `BOOST_FOREACH` are removed from code... It is a temporary solution, better than including the boost header and better than removing all `BOOST_FOREACH` in a big PR

**Comment by pwnorbitals** [comment]
imo, it does not add confusion as soon as it's well-commented, as I tried to do. Also, I tried removing all `BOOST_FOREACH` which reside in headers, but it would have resulted in a very big PR. This solution removes the dependency and doesn't require code changes.

**Comment by iFoggz** [comment]
its not hard to convert to for auto as i've started to do so myself in a branch. I like auto no need for PAIR_TYPES

**Comment by denravonska** [comment]
Superseded by #624. Might as well bite the bullet when we're just about to merge to staging with no large open PRs.

**Pull request closed without merging**